source/features/deep-blame.tsx

```diff
@@ -0,0 +1,140 @@
+								nodes {
+									commit {
+										oid
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	`);
+
+	const {nodes} = repository.object.associatedPullRequests;
+	const mergeCommit = nodes[0].mergeCommit.oid;
+	const pullRequestNumber = nodes[0].number;
+
+	if (pullRequestNumber === prNumber && mergeCommit === commit) {
+		return nodes[0].commits.nodes[0].commit.oid;
+	}
+
+	return false;
+}, {
+	cacheKey: ([commit]) => __featureName__ + commit
+});
+
+function getCommitHash(commit: HTMLElement): string {
+	return (commit.nextElementSibling! as HTMLAnchorElement).href.split('/').slice(-1)[0];
+}
+
+const githubSpinner = (
+	<img
+		src="https://github.githubassets.com/images/spinners/octocat-spinner-128.gif"
+		alt="Octocat Spinner Icon"
+		className="mr-2"
+		width="18"
+	/>
+);
+
+async function redirectToBlameCommit(event: DelegateEvent<MouseEvent, HTMLLinkElement>): Promise<void | false> {
+	const blameLink = event.delegateTarget;
+	const {lineNumber, commit: pullRequestCommit, prNumber} = blameLink.dataset;
+	if (blameLink.href && event.altKey) {
+		event.preventDefault();
+	}
+
+	blameLink.firstElementChild!.replaceWith(githubSpinner);
+	// Hide tooltip after click, it’s shown on :focus
+	blurAccessibly(blameLink);
+
+	const prBlameCommit = await getPullRequestBlameCommit(pullRequestCommit!, Number(prNumber));
+	if (!prBlameCommit) {
+		// Restore the regular version link if there was one
+		if (blameLink.href) {
+			blameLink.setAttribute('aria-label', 'View blame prior to this change.');
+			blameLink.classList.remove('rgh-deep-blame');
+			blameLink.firstElementChild!.replaceWith(versionIcon());
+		} else {
+			blameLink.firstElementChild!.remove();
+		}
+
+		alert('Unable to find linked Pull Request');
+		return;
+	}
+
+	const href = new URL(location.href.replace(getReference()!, prBlameCommit));
+	href.hash = 'L' + lineNumber!;
+	location.href = String(href);
+}
+
+async function init(): Promise<void | false> {
+	const pullRequests = select.all('[data-hovercard-type="pull_request"]');
+	if (pullRequests.length === 0) {
+		return false;
+	}
+
+	for (const pullRequest of pullRequests) {
+		const currentParentElement = pullRequest.closest('.blame-hunk')!;
+		const versionsParent = select('.blob-reblame', currentParentElement);
+		const currentLineNumber = select('.js-line-number', currentParentElement)!.textContent!;
+		const prNumber = select('.issue-link', currentParentElement)!.textContent!.replace('#', '');
+		const commitHash = getCommitHash(pullRequest);
+
+		if (select.exists('a', versionsParent!)) {
+			const versionsElement = select('a', versionsParent!)!;
+			versionsElement.setAttribute('aria-label', 'View blame prior to this change. Press Alt to view the `deep-blame`');
+			versionsElement.classList.add('rgh-deep-blame');
+			versionsElement.dataset.commit = commitHash;
+			versionsElement.dataset.lineNumber = currentLineNumber;
+			versionsElement.dataset.prNumber = prNumber;
```

Comment:
Why add these here (and in the link below)? They can just be found inside `redirectToBlameCommit`

In this function you just need to change the label (or add the icon)

Reply:
When I am doing the loop I am looking at the pull request number. The version element is not within the same div. So to get the info I would need to do 2x closest. Which is why I saved everything as a dataset

Reply:
There’s no way you have to do closest twice. Closest is towards the closest common ancestor, by definition once you found one, the others are also common.

Reply:
You are correct. I did not word it correctly.

The pull request popup ( this is the one were scanning for in the beginning) has all the elements we need right next to it. `('.blame-hunk')` 
Otherwise I would need to go up to the parentElement and then to `('.blame-hunk')` and get all the info.

I you feel that is better, I will change it.

Reply:
> I you feel that is better, I will change it.

Yes